core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -955,6 +977,11 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * The variable will be sent to each cluster only once.
    */
   def broadcast[T: ClassTag](value: T): Broadcast[T] = {
+    assertNotStopped()
+    if (classOf[RDD[_]].isAssignableFrom(classTag[T].runtimeClass)) {
```

Comment:
Actually, maybe this check should go somewhere else, since I think that it might technically have been safe to _create_ a broadcast variable with an RDD, even though doing anything with it would trigger errors.